test/utils.spec.js

```diff
@@ -76,33 +76,49 @@ describe('unit tests for util functions', () => {
 
   describe('object to event function', () => {
     it('should not accept invalid parameter', async () => {
-      expect(objectToEvent({})).to.be.instanceOf(ExtendedEvent);
-      expect(objectToEvent({ type: 'foo' })).to.be.instanceOf(ExtendedEvent);
+      const o = objectToEvent({});
+      expect(o).to.be.instanceOf(ExtendedEvent);
+      const type = o.getType();
+      expect(type).to.be.undefined;
```

Comment:
The linter doesn't seem to like this. I think `expect(type).to.be.equal(undefined);` would work